nixos/modules/services/security/vaultwarden/default.nix

```
@@ -191,8 +197,36 @@ in {
         ProtectHome = "true";
         ProtectSystem = "strict";
         AmbientCapabilities = "CAP_NET_BIND_SERVICE";
-        StateDirectory = "bitwarden_rs";
+        StateDirectory = (
+          if lib.versionAtLeast config.system.stateVersion "24.05"
+          then "vaultwarden"
+          else "bitwarden_rs"
+        );
         StateDirectoryMode = "0700";
+        RemoveIPC = true;
+        NoNewPrivileges = true;
+        CapabilityBoundingSet = "";
+        SystemCallFilter = [ "@system-service" ];
+        ProtectProc = "invisible";
+        ProtectClock = true;
+        ProcSubset = "pid";
+        PrivateUsers = true;
+        ProtectHostname = true;
+        ProtectKernelTunables = true;
+        RestrictAddressFamilies = [
+          "AF_INET"
+          "AF_INET6"
+          "AF_NETLINK"
```

```suggestion
          "AF_NETLINK"
          "AF_UNIX"
```
otherwise connecting to postgres via unix socket fails with:

```
"connection to server on socket \"/run/postgresql/.s.PGSQL.5432\" failed: could not create socket: Address family not supported by protocol\n",
```